### PR TITLE
Added file extensions config parameter

### PR DESCRIPTION
**Pull request by samskiter**
This adds the ability to ignore certain file extensions.

It's something I need personally, but I thought I would show you it and see what you think to including it upstream...?

### CODE REVIEW

**Comment by samskiter** [comment]
Note that this will conflict with https://github.com/joelittlejohn/jsonschema2pojo/pull/547 if merged. Easy fix though...

**Comment by samskiter** [comment]
Hmm, I didn't receive an error when running the tests, but when running `mvn install` I'm getting an error in the gradle plugin:

```
Caused by: java.lang.AbstractMethodError: org.jsonschema2pojo.gradle.JsonSchemaExtension_Decorated.getFileExtensions()[Ljava/lang/String;
	at org.jsonschema2pojo.Jsonschema2Pojo.getNodeName(Jsonschema2Pojo.java:147)
	at org.jsonschema2pojo.Jsonschema2Pojo.generateRecursive(Jsonschema2Pojo.java:109)
	at org.jsonschema2pojo.Jsonschema2Pojo.generate(Jsonschema2Pojo.java:73)
	at org.jsonschema2pojo.Jsonschema2Pojo$generate.call(Unknown Source)
	at org.jsonschema2pojo.gradle.GenerateJsonSchemaJavaTask.generate(GenerateJsonSchemaJavaTask.groovy:65)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:63)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:218)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:211)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:200)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
```

I'm totally at a loss when it comes to groovy. can anyone help?

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/Jsonschema2Pojo.java [hunk 1]
```diff
@@ -141,9 +141,29 @@ private static Annotator getAnnotator(GenerationConfig config) {
...
+            boolean extensionRemoved = false;
+            for (int i = 0; i < extensions.length; i++) {
+                String extension = extensions[i];
+                if (extension.length() == 0) {
+                    continue;
+                }
+                if (!extension.startsWith(".")) {
+                    extension = "." + extension;
+                }
+                if (fileName.endsWith(extension)) {
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/Jsonschema2Pojo.java]: I think `removeEnd` from StringUtils could be used here.

**Comment by samskiter** [comment]
Just updated to fix the conflict. Do you have any idea why this won't build @joelittlejohn ? Something Ive done wrong with groovy types?

**Comment by samskiter** [comment]
perhaps @ben-manes might know? sorry to bug you but I'm totally stumped - `String[] fileExtensions` should create an implementation of the getter `public String[] getFileExtensions()` in the interface right?

**Comment by ben-manes** [comment]
That's what I thought and is described that way in [Section 1.6.2](http://groovy-lang.org/objectorientation.html#fields). Maybe its something weird about array handling?

**Comment by ben-manes** [comment]
Can you just define an explicit getter? Its probably not worth fighting with the compiler for something this small.

**Comment by joelittlejohn** [comment]
Travis seems perfectly happy with this. Could it be a problem in your local m2 repo? Maybe some bad state is not being cleared in your local project, have you tried a fresh clone?

**Comment by samskiter** [comment]
Travis runs `mvn verify` which worked fine for me too. but `mvn install` threw the error. I'll clear my repo out and have another go...

**Comment by samskiter** [comment]
@joelittlejohn spot on point with the m2 repo - cleared and it's working now. That's frustrating.

are you interested in merging this? I'll look at using `removeEnd` now.

**Comment by samskiter** [comment]
used removeEnd

**Comment by samskiter** [comment]
@joelittlejohn If this is something you are interested in, it's good to go as far as I'm concerned... If not, I'll close and just maintain in my fork.

**Comment by joelittlejohn** [comment]
Hi Sam. Sorry, I haven't had much time to dedicate to this project recently.

One thing that struck me after thinking about this a bit further: Would it
be better to support arbitrary replacements in the name? Basically a
similar feature, but not restricted to the end of the filename. In your
case you want to find '.schema' and replace it with an empty string.

**Comment by samskiter** [comment]
Hey, no worries. I just happen to have a chance to put a bit into this to
get what we need from it (I should have a good chunk more time to put in
soon).

We could go for wildcards/regex if you like?

**Comment by samskiter** [comment]
Any decision on this? I'm starting to get quite far ahead of you on my branches and I don't want to end up with a fork

**Comment by joelittlejohn** [comment]
Lets do this! 👍

**Pull request merged**